src/components/d3/AxisGroup.js

```diff
@@ -0,0 +1,47 @@
+const React = require('react');
+const d3 = require('d3');
+
+const ChartUtils = require('../../utils/Chart');
+
+const AxisGroup = React.createClass({
+  propTypes: {
+    axis: React.PropTypes.string.isRequired,
+    axisFormatFunction: React.PropTypes.func.isRequired,
+    data: React.PropTypes.array.isRequired,
+    orientation: React.PropTypes.string.isRequired,
+    scaleFunction: React.PropTypes.func.isRequired,
+    translation: React.PropTypes.string
+  },
+
+  getDefaultProps () {
+    return {
+      translation: 'translate(0,0)'
+    };
+  },
+
+  componentDidMount () {
+    this._renderAxis();
+  },
+
+  componentDidUpdate () {
+    this._renderAxis();
+  },
+
+  _renderAxis () {
+    const tickValues = ChartUtils.getAxisTickValues(this.props.data, this .props.axis);
+    const axisFunction = d3.svg.axis()
+      .scale(this.props.scaleFunction())
```

Comment:
Did you want to hoist these `this.props` at the top? Or just the long ones? Asking because we want to be consistent.